components/switch/src/switch.js

```diff
@@ -0,0 +1,165 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+import { SwitchText, SwitchToggle } from './styled';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const Switch = ({
+	name,
+	id,
+	toggleText,
+	size,
+	block,
+	flip,
+	srOnlyText,
+	defaultChecked,
+	disabled,
```

Comment:
we should also be taking an onToggle prop to allow for the app to control the state of this switch (controlled components paradigm)